src/ImageManipulation.php

```diff
@@ -709,6 +711,33 @@ public function ThumbnailURL($width, $height)
         return $this->getIcon();
     }
 
+    /**
+     * Convert the file to another format if there's a registered converter that can handle it.
+     *
+     * @param string $toFormat The file extension you want to convert to - e.g. "webp".
+     * @param bool $returnNullOnFailure If true and the conversion fails, `null` is returned instead.
+     * This is useful if calling this method from a template, for example, where you don't want to be throwing
+     * exceptions for end users.
+     * @throws FileConversionException If the conversion fails and $returnNullOnFailure is false.
+     */
+    public function convert(string $toFormat, bool $returnNullOnFailure = false): ?AssetContainer
```

Comment:
Note that this is in the `ImageManipulation` trait because this trait is shared in both the `File` and `DBFile` class. The issue originally had some comments about this that I wanted to have as part of the ACs but it was decided that they [were not relevant](https://github.com/silverstripe/silverstripe-assets/issues/489#issuecomment-1945698768) - so we're stuck with putting a file-thing into an image trait.